WrightTools/data.py

```diff
@@ -2861,9 +2861,79 @@ def from_shimadzu(filepath, name=None, verbose=True):
     return data
 
 
+def from_1130(filepath, name=None, delimiter=None, verbose=True):
+    """Create a data object from a SPC-130 TCSPC an exported
+    comma-delimited (.asc) file within the SPCM 9.75 software.
+
+    Parameters
+    ----------
+    filepath : string
+        Path to SPC-130 output file (.asc).
+    name : string (optional)
+        Name to give to the created data object. If None, filename is used.
+        Default is None.
+    verbose : boolean (optional)
+        Toggle talkback. Default is True.
+
+    Returns
+    -------
+    WrightTools.data.Data object
+    """
+    # check filepath
+    if not os.path.isfile(filepath):
+        raise wt_exceptions.FileNotFound(path=filepath)
+    if not filepath.endswith('asc'):
+        wt_exceptions.WrongFileTypeWarning.warn(filepath, 'asc')
+    # set unspecified delimiter parameter arg to ','
+    if delimiter is None:
+        delimiter = ','
+    else:
+        pass
+    # import data
+    # now import file as a local var as comma-delimited .asc file
+    arr = np.genfromtxt(filepath,
+                        skip_header=10, skip_footer=1,
+                        delimiter=delimiter).T
+    # unexpected delimiter handler
+    if np.any(np.isnan(arr)):
+        # delimiter warning dictionary
+        delim_args = [',', '', '\t', ';', ':']
+        delim_strs = ['comma', 'space', 'tab', 'semicolon', 'colon']
+        delim_dict = dict(zip(delim_args, delim_strs))
+        if verbose:
+            print("Error: file is not %s-delimited!\n"
+                  "Trying other delimiters "
+                  "in wt.data.from_spc130() call." % delim_dict[delimiter])
+        for delimiter in delim_args:
+            arr = np.genfromtxt(filepath,
+                                skip_header=10, skip_footer=1,
+                                delimiter=delimiter).T
+            if not np.any(np.isnan(arr)):
+                if verbose:
+                    print("Error resolved: file is %s-delimited."
+                          % delim_dict[delimiter])
+                break
```

Comment:
personally I would use warnings as opposed to print conditional to `verbose` in this if block

Reply:
I agree, here, warnings are here for just such an occasion (though I may put the resolved in a logging layer like INFO, if we had such logging implemented)